nsqlookup/resolver.go

```diff
@@ -134,9 +133,78 @@ func (r *ConsulResolver) Resolve(ctx context.Context) (list []string, err error)
 		address = "http://" + address
 	}
 
-	if req, err = http.NewRequest("GET", address+"/v1/catalog/service/"+service, nil); err != nil {
+	// get list of check results for service
+	var checksResults []struct {
+		Node string
+	}
+
+	err = r.getConsul(ctx, fmt.Sprintf("v1/health/checks/%s?passing", service), &checksResults)
+	if err != nil {
 		return
 	}
+
+	// get list of nodes for service
+	var serviceResults []struct {
+		Node           string
+		Address        string
+		ServiceAddress string
+		ServicePort    int
+	}
+
+	err = r.getConsul(ctx, fmt.Sprintf("v1/catalog/service/%s", service), &serviceResults)
```

Comment:
For a simple concatenation you're better off using the `+` operator, the compiler will generate much faster code than using the `fmt` package which needs to use reflection on the formatted values.